### PR TITLE
kv patch command

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

File: command/kv_patch.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+type KVPatchCommand struct {
+	*BaseCommand
+
+	flagCAS   int
+	testStdin io.Reader // for tests
+}
+
+func (c *KVPatchCommand) Synopsis() string {
+	return "Sets or updates data in the KV store without ovewriting."
```
Comment (briankassouf) [inline command/kv_patch.go]: overwriting*

File: command/kv_patch.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+	"github.com/posener/complete"
+)
+
+var _ cli.Command = (*KVPatchCommand)(nil)
+var _ cli.CommandAutocomplete = (*KVPatchCommand)(nil)
+
+type KVPatchCommand struct {
+	*BaseCommand
+
+	flagCAS   int
```
Comment (briankassouf) [inline command/kv_patch.go]: I don't think this is getting used anywhere. I wonder if we should allow a cas flag that verifies the retrieved version equals that flag

Comment (jefferai) [reply command/kv_patch.go]: We could, although in that case you might as well have just modified the output of that initial read. Could defer until someone asks.

Comment (briankassouf) [reply command/kv_patch.go]: True, i'm fine with that

**Comment by jantman** [comment]
:clap:

File: command/kv_patch.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+	testStdin io.Reader // for tests
+}
+
+func (c *KVPatchCommand) Synopsis() string {
+	return "Sets or updates data in the KV store without overwriting."
+}
+
+func (c *KVPatchCommand) Help() string {
+	helpText := `
+Usage: vault kv put [options] KEY [DATA]
```
Comment (kalafut) [inline command/kv_patch.go]: put -> patch

**Pull request merged**